### PR TITLE
Replacing afl-unicorn with unicornafl, mostly rewritten.

### PR DESCRIPTION
**Pull request by domenukk**
This PR will replace afl-unicorn with a major rewrite.
This breaks the "api": scripts have to manually start the forkserver. 
This makes the fuzzer far more useful.
Also, writing a unicorn fuzzer in C is now viable.

I'd be glad if somebody tested this before merge :)
The diffs give me headaches.

### CODE REVIEW

File: unicorn_mode/patches/afl-unicorn-tcg-runtime-inl.h [hunk 1]
```diff
@@ -33,78 +33,6 @@
...
-        }
-
-      }
-
-    }
-
-  }
-
-}
-
```
Comment (domenukk) [inline unicorn_mode/patches/afl-unicorn-tcg-runtime-inl.h]: Not sure where the tcg helper went along the way. I was under the impression compcov is still in. Maybe it went somewhere else - I'll have to double check.

**Comment by domenukk** [comment]
I would redo this PR, depending on the decisions in https://github.com/vanhauser-thc/AFLplusplus/issues/117

**Comment by vanhauser-thc** [comment]
@domenukk we have a unanimous decision in #117 to go for cloning the repo :)

**Comment by domenukk** [comment]
All right, great. @vanhauser-thc if you fork https://github.com/unicorn-engine/unicorn I'll open a PR there for unicornafl and open a new PR here for the repo integration afterwards.
If anything else comes up, let me know.

**Comment by vanhauser-thc** [comment]
Here you go: https://github.com/vanhauser-thc/unicorn

**Pull request closed without merging**